ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -190,26 +188,31 @@ public boolean accept(File dir, String name) {
       log.setActiveSegment(getCurrentEndOffset().getName());
       logEndOffsetOnStartup = log.getEndOffset();
 
-      // After recovering the last messages, and setting the log end offset, let the hard delete thread do its recovery.
-      // NOTE: It is safe to do the hard delete recovery after the regular recovery because we ensure that hard deletes
-      // never work on the part of the log that is not yet flushed (by ensuring that the message retention
-      // period is longer than the log flush time).
-      logger.info("Index : " + datadir + " Starting hard delete recovery");
-      hardDeleter.performRecovery();
-      logger.info("Index : " + datadir + " Finished performing hard delete recovery");
+      if (hardDelete != null) {
```

Comment:
The compaction target log and index will be created by the data mover and discarded at the end of a single cycle of compaction. Indexes created for this purpose do not need a hard delete component.